GVFS/GVFS/CommandLine/CloneVerb.cs

```diff
@@ -121,7 +121,8 @@ public override void Execute()
                 CacheServerInfo cacheServer = null;
                 ServerGVFSConfig serverGVFSConfig = null;
 
-                using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "GVFSClone"))
+                string mountId = this.NoMount ? null : GVFSVerb.CreateMountId();
+                using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "GVFSClone", null, mountId))

```

Comment:
@sanoursa good question.  we were thinking it would be useful to know if the mount was initiated from a clone.  Aka we did a clone and the corresponding mount failed.  Thoughts?

Reply:
MountVerb already knows (sort of) that it's being called by CloneVerb, and we could make that more explicit. If we need this info, I would rather have MountVerb's mount event include it. That being said, what useful information do we think we'll get from that? We can already tell the first mount event from an enlistmentId. Is it actually relevant if the user typed in `--no-mount` or not? And I'd bet that 99% of the calls to `--no-mount` come from our immediate team, not from any real customers.

Reply:
> We can already tell the first mount event from an enlistmentId.
 I'm not sure how we could that.  Can you please you elaborate?  

I think I can make the change less invasive by adding a true/false flag for starting from clone.  I'm not sure if we actually care what the original clone was, but rather is there a higher correlation of mount fails coming from clones.

Reply:
I don't know kusto syntax well enough, but if you gather all the mounts for an enlistment id, can't you just pick the first one as the "mount during clone"?

Reply:
>MountVerb already knows (sort of) that it's being called by CloneVerb, and we could make that more explicit.

> I think I can make the change less invasive by adding a true/false flag for starting from clone. I'm not sure if we actually care what the original clone was, but rather is there a higher correlation of mount fails coming from clones.

We already have `SkipMountedCheck` and `SkipVersionCheck` in `MountVerb` which are only set to true when `CloneVerb` calls `MountVerb` (I think this is what @sanoursa was referring to when he said the MountVerb sort of knows already).  Rather than adding another flag we could rename one of these to make it more explicit that the clone started the mount.

> I don't know kusto syntax well enough, but if you gather all the mounts for an enlistment id, can't you just pick the first one as the "mount during clone"?

That might not account for the `gvfs clone --no-mount`, but I'm not sure how frequent that is (note that we do log if clone was run with `--no-mount` and so I'd imagine we could adjust the query to account for that as well if we wanted to).

Reply:
@kewillford @wilbaker @sanoursa thanks for your input here.  

Thinking about the comments, I'm thinking of merging SkipMountedCheck/SkipVersionCheck into InitiatedByClone and passing that on to the mounting process for telemetry.  If we think we'll have a need for either of the Skips outside of Clone let me know.  This would negate the need for enlistmentId/mountId on the CloneVerb.

> I don't know kusto syntax well enough, but if you gather all the mounts for an enlistment id, can't you just pick the first one as the "mount during clone"?

I think that is a resonable approach.  If we don't stick enlistmentId into the Clone it isn't a simple join to find the Clone parameters, however I don't think we'll commonly need to join.  We can query the data by timestamp for a given enlistment.

Reply:
It would be nice if we could get an enlistmentId in the clone verb so that if someone is running multiple clones at the same time we would be able to match the clone log entries with the mount.  I don't think it is super critical because it probably doesn't happen that often.  It would just be easier to run a query the get all the events for a single enlistment or if we wanted to get the clone events for some event that has the enlistmentId.

Reply:
@kewillford we do get the EnlistmentRoot on both so we can match it up that way.  EnlistmentId is for sure a bulletproof way, but if we think the case in general isn't super common EnslitmentRoot should suffice.

Reply:
The EnlistmentRoot is added to every event that clone could log?  I see events that only have the DurationMs.  Not sure what errors during clone would include.

Reply:
@kewillford please forgive me if i'm telling you things you already know.

The way I think about it is that clone as a start and end event based on [this using block](https://github.com/Microsoft/VFSForGit/blob/f487e7f78df5493be5efe2e9a043a2c37fb5940c/GVFS/GVFS/CommandLine/CloneVerb.cs#L124).  The WriteStartEvent logs with enlistmentRoot.  When dispose is called on the using it creates a new event with just duration.  I'm going to push up a .md with the queries I'm thinking would work later today.  Once that is up I'd love your input on if they are sufficient.

Reply:
Sounds good thanks! I was just looking at the data in Kusto using 
```
let start = ago(1d);
MicrosoftGit
| where timestamp > start
| where name == "Microsoft.Git.GVFS.GVFSClone"
| extend d = parse_json(tostring(parse_json(data).Json))
```
And wasn't seeing anything that would tie data of running a `gvfs clone` together.  There is some `device_` and `user_` data but if I run clone 10 times on my machine because it is failing it gets more complicated.  We could use time to know the order of events unless I am running 2 at the same time.  Most of these are just contrived scenarios but would be easily handled with an enlistmentId on every event.

Reply:
@kewillford I've added a .md file with queries.  The scenario you describe is correct.  I can find the clone start event which has the parameters it ran with, however errors from clone until Mount you can only use a user_id and and timestamps on.  If we think it's a necessary scenario I can look into it for sure, but it would be some work to get enlistmentId on the events that come up (PreMount/etc..).